domains/runtime/evm/src/lib.rs

```diff
@@ -511,15 +500,12 @@ impl pallet_evm::OnChargeEVMTransaction<Runtime> for EVMCurrencyAdapter {
         base_fee: U256,
         already_withdrawn: Self::LiquidityInfo,
     ) -> Self::LiquidityInfo {
-        let tip =
-            <InnerEVMCurrencyAdapter as pallet_evm::OnChargeEVMTransaction<
-                Runtime,
-            >>::correct_and_deposit_fee(who, corrected_fee, base_fee, already_withdrawn);
-        if let Some(t) = tip.as_ref() {
-            // Record the evm transaction tip
-            OperatorRewards::note_operator_rewards(t.peek())
-        }
-        tip
+        // Record the evm actual transaction fee
+        OperatorRewards::note_operator_rewards(corrected_fee.as_u128());
+
+        <InnerEVMCurrencyAdapter as pallet_evm::OnChargeEVMTransaction<
+            Runtime,
+        >>::correct_and_deposit_fee(who, corrected_fee, base_fee, already_withdrawn)
```

Comment:
In main, the evm txn fee is handled in a way that follows the exact behavior of `frontier` which can be verbose but should be correct I think.

And the new change seems not to consider the case the `already_withdrawn` is `None` (though I'm not sure when it will be `None`):
https://github.com/subspace/frontier/blob/2b09a676cac8cca665c882a28bd13b2acef39395/frame/evm/src/lib.rs#L826-L832

Reply:
> In main, the evm txn fee is handled in a way that follows the exact behavior of frontier which can be verbose but should be correct I think.

No, that is incorrect. It is passing base_fee on every empty bundle which should not be case.
We only want the actual deposited fee from the address to be noted. I have verified this locally

> And the new change seems not to consider the case the already_withdrawn is None (though I'm not sure when it will be None):
This is None for XCM related and not in our use case

Reply:
> No, that is incorrect. It is passing base_fee on every empty bundle which should not be case.

But we are only noting evm txn fee after executing the evm txn (i.e. `post_dispatch`) and `submit_bundle` is not an extrinsic of the evm domain chain, when processing the empty bundle there should be no txn fee being noted IIUC, perhaps the inherit extrinsic?

> This is None for XCM related and not in our use case

It would be more robust if we handled the case here, non-blocker.

Reply:
> But we are only noting evm txn fee after executing the evm txn (i.e. post_dispatch) and submit_bundle is not an extrinsic of the evm domain chain, when processing the empty bundle there should be no txn fee being noted IIUC, perhaps the inherit extrinsic?

But that is not the case. Please look a Gemini-3f for empty bundles and you will notice the base fee is noted even for empty bundles

Reply:
Gemini-3f domain doesn't contain the `pallet-operator-rewards` and is using a fixed block reward. It does confuse me now, with the `None` changes the behavior of this PR and the main branch should be the same because the `base_fee` argument is not used directly:
https://github.com/subspace/frontier/blob/2b09a676cac8cca665c882a28bd13b2acef39395/frame/evm/src/lib.rs#L869-L872
https://github.com/paritytech/polkadot-sdk/blob/c54ea64af43b522d23bfabb8d917a490c0f23217/substrate/frame/balances/src/impl_currency.rs#L96-L102

Reply:
BaseFee is not meant to be used directly and is used internally to calculate the actual fee for a given txn. So `corrected_fee` already includes `base_fee` + `priority_fee` if there is any. More on this here - https://github.com/subspace/frontier/blob/2b09a676cac8cca665c882a28bd13b2acef39395/frame/evm/src/runner/stack.rs#L154-L167

This PR essentially uses the already correct_fee instead of noting base_fee and tip seperately. One way to verify this is  check the fee withdrawn from account matches the operator_rewards noted for this txn.

Reply:
> This PR essentially uses the already correct_fee instead of noting base_fee and tip separately.

I'm okay with the changes as it does make the code cleaner, but it does confuse me about the changes also fix an issue because from my understanding `corrected_fee == base_fee  + tip` so logically there is no difference between noting the `corrected_fee` and noting both `base_fee` and `tip` separately. Just want to make sure we are fixing the right issue.

> One way to verify this is check the fee withdrawn from account matches the operator_rewards noted for this txn.

There is already a test in main to ensure this:
https://github.com/subspace/subspace/blob/606d2592f78569dcfaebedb2694b6df49e0c13fc/domains/client/domain-operator/src/tests.rs#L2124-L2125

Reply:
> I'm okay with the changes as it does make the code cleaner, but it does confuse me about the changes also fix an issue because from my understanding corrected_fee == base_fee  + tip so logically there is no difference between noting the corrected_fee and noting both base_fee and tip separately.

That is point of the PR and mentioned in the description. Maybe I should have worded the PR correctly. I would prefer we not re-doing the calculation seperately and just rely on what is given from the frontier itself

Reply:
Okay, looks good to me for this PR as pure refactoring.

> I would prefer we not re-doing the calculation seperately and just rely on what is given from the frontier itself

BTW, handling base_fee and tip separately is also what provided by the frontier interface.

Reply:
> BTW, handling base_fee and tip separately is also what provided by the frontier interface.

I'm not sure why, but that brings in more complexity of fee calculation. We do not need to use just because its provided though 😉